src/modules/date/date-string.transformations.ts

```diff
@@ -29,4 +29,24 @@ export class DateStringTransformations {
     }
     return dateOnlyString + 'T00:00:00Z';
   }
+
+  getEffectiveDate(dateOnlyString: DateString): DateString {
+    const today = new Date();
+    if (new Date(dateOnlyString) < today) {
+      return this.addTimeToDateOnlyString(dateOnlyString);
+    } else {
+      const todayAsDateOnlyString = this.removeTimeIfExists(today.toISOString());
+      return this.addTimeToDateOnlyString(todayAsDateOnlyString);
+    }
+  }
```

Comment:
We have a function called `getEarliestDateFromTodayAnd` in `current-date.provider.ts` that can be used for part of this but maybe it would be easier to use in the `DateTransformations` - what do you think?

Either way, we shouldn't call this `getEffectiveDate` if we keep it here because it's not very clear in the context of this class _why_ this logic would get an effective date (or what an effective date even is). I think `getEarliestDateOnlyStringFromTodayAnd(otherDateOnlyString: DateOnlyString): DateOnlyString` would make more sense if it stays here